basic/src/tags/components/CreateTagDialog.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { Button } from "../../shared/components/Button";
+import { PortalDialog } from "../../shared/components/Dialog";
+import { CREATE_TAG_FORM_ID, CreateTagForm } from "./CreateTagForm";
+
+export function CreateTagDialog() {
+  const [tagDialogOpen, setTagDialogOpen] = React.useState(false);
+
+  return (
+    <>
+      <Button type="button" size="sm" onClick={() => setTagDialogOpen(true)}>
+        <span>Add a Tag</span>
+        <span>+</span>
+      </Button>
+      {tagDialogOpen && (
+        <PortalDialog
+          open={tagDialogOpen}
+          onClose={() => setTagDialogOpen(false)}
+        >
+          <section className="card relative flex max-w-lg flex-col">
+            <header className="px-6 pb-4 pt-8">
+              <h2 className="text-xl font-semibold">Create a new tag</h2>
+            </header>
+            <div className="overflow-y-auto p-4 px-6">
+              <CreateTagForm onTagCreated={() => setTagDialogOpen(false)} />
+            </div>
+            <footer className="flex justify-end gap-2 px-6 pb-8 pt-4">
+              <Button form={CREATE_TAG_FORM_ID} type="submit">
+                Create
+              </Button>
+              <Button
+                form={CREATE_TAG_FORM_ID}
+                type="button"
+                onClick={() => setTagDialogOpen(false)}
+                variant="danger"
+              >
+                Cancel
+              </Button>
+            </footer>
+          </section>
```

Comment:
```suggestion
          <section className="card relative flex max-h-full flex-col overflow-auto">
            <header>
              <h2 className="text-xl font-semibold">Create a new tag</h2>
            </header>
            <div className="overflow-y-auto mb-12 mt-8">
              <CreateTagForm onTagCreated={() => setTagDialogOpen(false)} />
            </div>
            <footer className="flex justify-end gap-2">
              <Button form={CREATE_TAG_FORM_ID} type="submit">
                Create
              </Button>
              <Button
                form={CREATE_TAG_FORM_ID}
                type="button"
                onClick={() => setTagDialogOpen(false)}
                variant="ghost"
              >
                Cancel
              </Button>
            </footer>
          </section>
```

1. Tried to clean it up a bit by using margins for creating space between elements instead of padding
2. Replaced the `danger` button for "Cancel" with a `ghost` button - the red was dominating and it wasn't that scary of an action - canceling a creation of a tag.

Reply:
Margin breaks the content for me
<img width="820" alt="image" src="https://github.com/user-attachments/assets/9276849b-f54a-4524-8fe3-3d857dfe81a0" />
Switching your margins to padding makes it work

And the content is also clipped 
<img width="86" alt="image" src="https://github.com/user-attachments/assets/d516dff3-935e-4209-b320-01996cc37637" />

Hm, there was a reason actually I used padding as I did, it's because I want the dialog header and footer to be always visible. The content of the dialog is expandable and scrollable. The padding is important for the placing of the scroll bar.
![image](https://github.com/user-attachments/assets/935b98d3-fd43-40ea-af72-0a031fe7bb26)
![image](https://github.com/user-attachments/assets/4550be8c-4bbf-494d-bd87-1ef2b9796087)

I should probably lower the padding though.

Reply:
Images a bit weird since I used the firefox shortcut to save the whole page instead of current screen.

Reply:
Yep, I think it makes sense to make the header and footer always visible. Just maybe reduce the padding and try to use as much space on mobile as possible so it's the best experience. I don't think we need to over-engineer the mobile styles, but I'm happy we are thinking about it :)